terraform/transform_flatten.go

```diff
@@ -53,6 +53,12 @@ func (t *FlattenTransformer) Transform(g *Graph) error {
 
 		// Go through the subgraph and flatten all the nodes
 		for _, sv := range subgraph.Vertices() {
+			// If the vertex already has a subpath then we assume it has
+			// already been flattened. Ignore it.
+			if _, ok := sv.(GraphNodeSubPath); ok {
```

Comment:
Yeah, I thought of that too, but I didn't want to risk missing adding that interface to every flat node. I'd rather do this and then just refactor later to have EVERY node have a `Path` and do it without flattening at all.